File: src/string/reverse_string/ReverseString.java
package string.reverse_string;

import java.util.Arrays;

public class ReverseString {
    static void reverseString(char[] s) {
        int start = 0;
        int end = s.length - 1;

        while (start < end) {
            char temp = s[start];
            s[start] = s[end];
            s[end] = temp;

            start++; end--;
        }
    }

    public static void main(String[] args) {
        char[] arr = {'a', 'b', 'c'};
        System.out.println("origin = " + Arrays.toString(arr));
        reverseString(arr);
        System.out.println("reversed = " + Arrays.toString(arr));
    }
}

File: src/string/reverse_string/ReverseString.kt
package string.reverse_string

import java.util.*

fun reverseString(s: CharArray): Unit {
    var start = 0;
    var end = s.size - 1;

    while (start < end) {
        s[start] = s[end].also { s[end] = s[start] }

        start++
        end--
    }
}

fun main() {
    val arr = charArrayOf('a', 'b', 'c')
    println("origin = ${arr.contentToString()}")
    reverseString(arr)
    println("reversed = ${arr.contentToString()}")
}